### PR TITLE
feat(ci): tag docker image as latest only for beta

### PR DESCRIPTION
**Pull request by shekhirin**
We need to tag as `latest` only Beta images, and not the Alpha images that we still release occasionally.

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -36,5 +43,9 @@ jobs:
         run: |
           docker run --privileged --rm tonistiigi/binfmt --install arm64,amd64
           docker buildx create --use --name cross-builder
+      - name: Build and push image, tag as "latest"
+        if: steps.check-latest.outputs.match == 'true'
```
Comment (onbjerg) [inline .github/workflows/docker.yml:47]:
```suggestion
        if: contains(github.event.ref, 'beta')
```

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -36,5 +43,9 @@ jobs:
...
+        run: make PROFILE=maxperf docker-build-push-latest
       - name: Build and push image
-        run: make PROFILE=maxperf docker-build-latest
+        if: steps.check-latest.outputs.match != 'true'
```
Comment (onbjerg) [inline .github/workflows/docker.yml:50]:
```suggestion
        if: contains(github.event.ref, 'beta')
```

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -23,6 +23,13 @@ jobs:
       packages: write
       contents: read
     steps:
+      - name: Check latest
+        id: check-latest
+        # Return `true` only if the tag matches the beta version regex
+        run: |
+          if [[ ${{ github.event.ref }} =~ ^refs/tags/v[0-9]+\.[0-9]+\.[0-9]+\.beta.*$ ]]; then
+              echo "match=true" >> $GITHUB_OUTPUT
+          fi
```
Comment (onbjerg) [inline .github/workflows/docker.yml:32]:
```suggestion
```

**Review by onbjerg — APPROVED** [review]
lgtm, suggestion on an alternative but up to you (dup code vs regex)

File: .github/workflows/docker.yml (hunk 1 above)
Comment (shekhirin) [reply .github/workflows/docker.yml]: yeah right, think that's enough

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**